src/System.IO.FileSystem/tests/File/Create.cs

```diff
@@ -152,16 +152,29 @@ public void OverwriteReadOnly()
         }
 
         [Fact]
+        [ActiveIssue("https://github.com/dotnet/corefx/issues/8655")]
         public void LongPath()
         {
             DirectoryInfo testDir = Directory.CreateDirectory(GetTestFilePath());
-            Assert.Throws<PathTooLongException>(() => Create(Path.Combine(testDir.FullName, new string('a', 300))));
 
-            //TODO #645: File creation does not yet have long path support on Unix or Windows
-            //using (Create(Path.Combine(testDir.FullName, new string('k', 257))))
-            //{
-            //    Assert.True(File.Exists(Path.Combine(testDir.FullName, new string('k', 257))));
-            //}
+            // Long path should throw PathTooLongException on Desktop and IOException
+            // elsewhere.
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<PathTooLongException>(
+                    () => Create(Path.Combine(testDir.FullName, new string('a', 300))));
+            }
+            else
+            {
+                Assert.Throws<IOException>(
+                    () => Create(Path.Combine(testDir.FullName, new string('a', IOInputs.MaxComponent + 1))));
+
+                var fileName = new string('k', IOInputs.MaxComponent - 1);
```

Comment:
Hm, doesn't seem like there is good infrastructure in place for building valid inputs to IO methods. I mostly was trying to uncomment the test validating long path support, but now thinking about it more, I don't think it exactly fits in this same test. Perhaps, I can revert to that commented out portion and once we get the CoreCLR bits merged in, I can work on expanding positive test cases for long path support throughout System.IO.* Thoughts on this?